core/blockchain.go

```diff
@@ -497,11 +509,31 @@ func (bc *BlockChain) cacheReceipts(hash common.Hash, receipts types.Receipts) {
 	bc.receiptsCache.Add(hash, receipts)
 }
 
-func (bc *BlockChain) cacheDiffLayer(diffLayer *types.DiffLayer) {
+func (bc *BlockChain) cacheDiffLayer(diffLayer *types.DiffLayer, sorted bool) {
```

Comment:
What is the purpose of `sorted` here?

Reply:
I will suggest passing `diffLayerCh` as a parameter of `cacheDiffLayer`,  and closing it once difflayer is cached. No bother to fetch it from cache.

Reply:
> I will suggest passing `diffLayerCh` as a parameter of `cacheDiffLayer`, and closing it once difflayer is cached. No bother to fetch it from cache.

fixed